app/src/main/java/intellistart/interviewplanning/model/booking/validation/BookingValidator.kt

```diff
@@ -48,41 +50,52 @@ class BookingValidator(
         }
     }
 
-    private fun checkBookingForOverlapsWithSlots(booking: Booking) {
-        val periodOfInterviewer: Period = slotService.getById(booking.interviewerSlotId).period
-        val periodOfCandidate: Period = slotService.getById(booking.candidateSlotId).period
-
+    private fun checkBookingForOverlapsWithSlots(booking: Booking): Mono<Unit> = Mono.zip(
+        slotService.getById(booking.interviewerSlotId),
+        slotService.getById(booking.candidateSlotId)
+    ).flatMap { tuple ->
+        val periodOfInterviewer: Period = tuple.t1.period
+        val periodOfCandidate: Period = tuple.t2.period
         val bookingPeriod: Period = booking.period
 
         if (
             !periodOfInterviewer.date.isEqual(periodOfCandidate.date) ||
             !periodService.isFirstInsideSecond(bookingPeriod, periodOfInterviewer) ||
             !periodService.isFirstInsideSecond(bookingPeriod, periodOfCandidate)
         ) {
-            throw BookingException(BookingExceptionProfile.SLOTS_NOT_INTERSECTING)
+            Mono.error(BookingException(BookingExceptionProfile.SLOTS_NOT_INTERSECTING))
+        } else {
+            Unit.toMono()
         }
     }
 
     private fun checkBookingForOverlapsWithOtherBookings(
-        booking: Booking
-    ) {
+        booking: Booking,
+    ): Mono<Unit> = Mono.zip(
+        slotService.getById(booking.interviewerSlotId),
+        slotService.getById(booking.candidateSlotId)
+    ).handle { (interviewerSlot, candidateSlot), sink ->
+        val interviewerBookings: Collection<Booking> = interviewerSlot.bookings
+        val candidateBookings: Collection<Booking> = candidateSlot.bookings
+
         validatePeriodNotOverlappingWithOtherBookingPeriods(
             booking.id,
             booking.period,
-            slotService.getById(booking.interviewerSlotId).bookings
+            interviewerBookings
         )
 
         validatePeriodNotOverlappingWithOtherBookingPeriods(
             booking.id,
             booking.period,
-            slotService.getById(booking.candidateSlotId).bookings
+            candidateBookings
         )
+        sink.complete()
```

Comment:
We have the next actions here:
1. get interviewer slot and validate with `validatePeriodNotOverlappingWithOtherBookingPeriods`
2. get candidate slot and validate with `validatePeriodNotOverlappingWithOtherBookingPeriods`

We can extract common part:
1. get slot by user id
2. validate the slot with `validatePeriodNotOverlappingWithOtherBookingPeriods`

So, how we can improve this code:
```kotlin
private fun validateBookingIsApplicableToUser(userId: String, booking: Booking): Mono<Unit> =
    slotService.getById(userId)
        .doOnNext { user ->
            validatePeriodNotOverlappingWithOtherBookingPeriods(
                booking.id,
                booking.period,
                user.bookings
            )
        }
        .thenReturn(Unit)
```

then update function `checkBookingForOverlapsWithOtherBookings`:
```kotlin
    private fun checkBookingForOverlapsWithOtherBookings(
        booking: Booking,
    ): Mono<Unit> = Mono.`when`(
        validateBookingIsApplicableToUser(booking.interviewerSlotId),
        validateBookingIsApplicableToUser(booking.candidateSlotId),
    ).thenReturn(Unit)
```

----

Maybe I've missed something, so correct me if I wrong